File: dotnet/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/DeleteTextCommand.cs
namespace CommandDesignPattern.Example1;

/// <summary>
/// Concrete Command - Delete text command
/// </summary>
public class DeleteTextCommand : ICommand
{
    private readonly TextEditor _editor;
    private readonly int _length;
    private string _deletedText = string.Empty;

    public DeleteTextCommand(TextEditor editor, int length)
    {
        _editor = editor;
        _length = length;
    }

    public void Execute()
    {
        // Store the deleted text for undo
        var contentLength = _editor.Content.Length;
        var deleteLength = Math.Min(_length, contentLength);
        
        if (deleteLength > 0)
        {
            _deletedText = _editor.Content.Substring(contentLength - deleteLength);
            _editor.DeleteText(deleteLength);
        }
    }

    public void Unexecute()
    {
        if (!string.IsNullOrEmpty(_deletedText))
        {
            _editor.AppendText(_deletedText);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/src/DeleteTextCommand.ts
import { ICommand } from "./ICommand";
import { TextEditor } from "./TextEditor";

/**
 * Concrete Command — deletes N characters from the end of the editor.
 */
export class DeleteTextCommand implements ICommand {
  private _deletedText = "";

  constructor(
    private readonly _editor: TextEditor,
    private readonly _length: number
  ) {}

  execute(): void {
    const deleteLength = Math.min(this._length, this._editor.content.length);
    if (deleteLength > 0) {
      this._deletedText = this._editor.content.slice(-deleteLength);
      this._editor.deleteText(deleteLength);
    }
  }

  unexecute(): void {
    if (this._deletedText) {
      this._editor.appendText(this._deletedText);
    }
  }
}